### DISCUSSION TITLE
Does `ruff` reimplement bugs or proactively complete feature requests?

### DISCUSSION THREAD

**Discussion by jamesbraza** [astral-sh/ruff#4143] [General]
Let's say `ruff` is reimplementing a `flake8` plugin that has known bugs.  Will `ruff` match the known bugs?

Similarly, if `ruff` sees an outstanding feature request collecting dust, might `ruff` complete it during the re-implementation process?

In general, I am wondering how close/far apart `ruff` can be from tools it reimplements.  Thank you!

**Comment by charliermarsh** [comment]
Good questions!

1. We don't guarantee "bug for bug" compatibility, so we _will_ deviate from "upstream" plugins if it means fixing bugs.
2. We typically avoid proactively implementing feature requests that deviate from "upstream" plugins. There may be a small number of exceptions (I can't think of any off-hand, but they might exist).
3. Not asked, but: we do strive to retain feature parity over time, even as the upstream plugins evolve. E.g., we've added rules recently to match the latest `flake8-bugbear` and `flake8-comprehensions` releases.

**Reply by jamesbraza** [reply]
Wonderful, these are great answers!

I asked because I open bug/feature requests in tooling repos, so I am wondering if I should duplicate them in `ruff`.  It sounds like the answer is no, the `ruff` community tracks new releases autonomously and matches them in time.
